pkg/descheduler/strategies/node_affinity.go

```diff
@@ -53,21 +63,52 @@ func removePodsViolatingNodeAffinityCount(ds *options.DeschedulerServer, strateg
 					if maxPodsToEvict > 0 && nodepodCount[node]+1 > maxPodsToEvict {
 						break
 					}
-					if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil && pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+					if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil {
```

Comment:
You can combine both the blocks under switch cases to function and re-use them.

Reply:
Got it.